snowpack/src/build/build-import-proxy.ts

```diff
@@ -208,15 +228,20 @@ export async function wrapImportProxy({
   return generateDefaultImportProxy(url);
 }
 
-const PUBLIC_ENV_REGEX = /^SNOWPACK_PUBLIC_/;
 export function generateEnvModule(mode: 'development' | 'production') {
+  const envObject = getSnowpackPublicEnvVariables();
+  envObject.MODE = mode;
+  envObject.NODE_ENV = mode;
+  return `export default ${JSON.stringify(envObject)};`;
+}
+
+const PUBLIC_ENV_REGEX = /^SNOWPACK_PUBLIC_/;
```

Comment:
Should it be `/^SNOWPACK_PUBLIC_.*/` so it doesn't match a `SNOWPACK_PUBLIC_` variable?

Reply:
yea, good catch. I think that that's worth adding. technically it should be `/^SNOWPACK_PUBLIC_.+/`, since * can match 0 items.

Reply:
Indeed, `/^SNOWPACK_PUBLIC_.+/` is what I meant 😉

Reply:
I'll make a follow up issue so we can ship this change